src/bugreporter/odemis_bugreporter.py

```diff
@@ -256,18 +256,17 @@ def compress_files(self):
                 models = glob(os.path.join(odemis_config['CONFIGPATH'], '*/*.odm.yaml'))
             files.extend(models)
 
-            # FIXME: the following two are not working properly if glob doesn't sort by time,
-            # which it probably doesn't do.
-
             # Add the latest overlay-report if it's possibly related (ie, less than a day old)
-            overlay_reps = glob(os.path.join(home_dir, 'odemis-overlay-report', '*'))
+            overlay_reps = glob(os.path.join(home_dir, 'odemis-overlay-report/*'))
+            # glob returns list of folders sorted by their names. The folder name corresponds
+            # to the date of the overlay report, so no further sorting is necessary
             if overlay_reps and (time.time() - os.path.getmtime(overlay_reps[-1])) / 3600 < 24:
-                files.append(overlay_reps[-1])
+                files.extend(glob(os.path.join(overlay_reps[-1], '*')))
 
             # Add the latest DELPHI calibration report if it's possibly related (ie, less than a day old)
-            delphi_calib_reps = glob(os.path.join(home_dir, 'delphi-calibration-report', '*'))
+            delphi_calib_reps = glob(os.path.join(home_dir, 'delphi-calibration-report/*'))
             if delphi_calib_reps and (time.time() - os.path.getmtime(delphi_calib_reps[-1])) / 3600 < 24:
-                files.append(delphi_calib_reps[-1])
+                files.extend(glob(os.path.join(delphi_calib_reps[-1], '*')))
```

Comment:
I'd suggest to just keep the name of the directory, and handle directory names below. It should be more future-proof (when we want to add more directories later on).

Reply:
Why is it more future proof? I bit earlier in the code, when we define the list of all other files, we also use the complete path for each file, so I thought it would be more consistent to also do it for the overlay/calibration reports.

Do you also want me to change this as well? I actually thought it was a good solution because it's simple and gives a zip file that is organized into folders which is nice.

```files = ['/var/log/odemis.log', os.path.join(home_dir, 'odemis-gui.log'),
                 os.path.join(home_dir, 'odemis-gui.log.1'), '/etc/odemis.conf', '/var/log/syslog',
                 os.path.join(home_dir, 'odemis-mic-selector.log'), '/tmp/odemis-bug-screenshot.png',
                 '/etc/odemis-settings.yaml']
```